lib/Cro/HTTP/ReverseProxy.pm6

```diff
@@ -0,0 +1,91 @@
+use Cro;
+use Cro::HTTP::Client;
+
+class Cro::HTTP::ReverseProxy does Cro::Transform {
+    has $.to;
+    has $.to-absolute;
+    has $!destination;
+    has $!client;
+    has $.request;
+    has $.response;
+    has %.ca;
+
+    submethod TWEAK() {
+        unless ($!to.defined ^^ $!to-absolute.defined) {
+            die 'Either `to` or `to-absolute` must be specified for ReverseProxy';
+        }
+        $!to = self!trim-url($!to) if $!to ~~ Str;
+        $!to-absolute = self!trim-url($!to-absolute) if $!to-absolute ~~ Str;
+        with $!to {
+            die '`to` of ReverseProxy must be a Str or code block' unless $!to ~~ Str|Callable;
+            $!destination = $!to;
+        }
+        else {
+            $!destination = $!to-absolute;
+        }
+
+        my %args = %!ca ?? (:%!ca, :http<1.1 2>) !! ();
+        $!client = Cro::HTTP::Client.new(|%args);
+    }
+
+    method consumes() { Cro::HTTP::Request }
+    method produces() { Cro::HTTP::Response }
+
+    method !trim-url($url) { $url.ends-with('/') ?? $url.substr(0, *-1) !! $url }
+
+    method transformer(Supply $pipeline --> Supply) {
+        supply whenever $pipeline -> $request {
+            my $*PROXY-STATE;
```

Comment:
I think we'll have to declare a leixcal `my $proxy-state` and this level, and bind it like `my $*PROXY-STATE := $proxy-state;` ahead of calls that should be able to see it in order to get it robustly available even over async boundaries.